pkg/generate/app/cmd/resolve.go

```diff
@@ -116,7 +119,7 @@ func Resolve(r *Resolvers, c *ComponentInputs, g *GenerationInputs) (*ResolvedCo
 	}
 
 	if g.Strategy != generate.StrategyUnspecified && len(repositories) == 0 && !g.BinaryBuild {
-		return nil, errors.New("when --strategy is specified you must provide at least one source code location")
+		return nil, errors.New("when --strategy is specified you must provide at least one source code location; see the argument classification results above for possible explanations a source code location was not found")
```

Comment:
Good point re: assured of explanations existing above.

That said, I think the current text is still lacking.

How about:

`--strategy is specified and none of the arguments provided could be classified as a source code location`